src/ChoiceFieldset/ChoiceFieldsetList.tsx

```diff
@@ -0,0 +1,66 @@
+import React, {ChangeEvent} from 'react'
+import styled from 'styled-components'
+import {CheckboxInputField, RadioInputField} from '..'
+import {get} from '../constants'
+import {uniqueId} from '../utils/uniqueId'
+import {Slot, ChoiceFieldsetContext} from './ChoiceFieldset'
+import ChoiceFieldsetListContext from './ChoiceFieldsetListContext'
+
+export interface ChoiceFieldsetListProps {
+  /**
+   * Whether multiple items or a single item can be selected
+   */
+  selectionVariant?: 'single' | 'multiple'
+}
+
+const List = styled.ul`
+  display: flex;
+  flex-direction: column;
+  list-style: none;
+  margin: 0;
+  padding: 0;
+
+  > li + li {
+    margin-top: ${get('space.2')};
+  }
+`
+
+const ChoiceFieldsetList: React.FC<ChoiceFieldsetListProps> = ({selectionVariant, children}) => (
+  <Slot name="ChoiceList">
+    {({name, onSelect, disabled, selected = []}: ChoiceFieldsetContext) => {
+      const getSelectedCheckboxes = (value: string, checked: boolean): string[] => {
+        if (checked) {
+          return selectionVariant === 'multiple' ? [...selected, value] : [value]
+        }
+
+        return selected.filter(selectedValue => selectedValue !== value)
+      }
+      return (
+        <ChoiceFieldsetListContext.Provider
+          value={{
+            disabled,
+            selected,
+            name: name || uniqueId(),
```

Comment:
Do we need to regenerate this unique id on each render? I think it be more performant to hoist this outside of the component render/return?

Reply:
I used a hook that doesn't calculate a new ID on each re-render.